packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -632,12 +661,14 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 
 		// Deal with transport compression
 		if (this.transportCompressionEnabled) {
+			// Each WS message received is a full gateway message for zstd streaming, but for zlib it's chunked
 			const flush =
-				decompressable.length >= 4 &&
-				decompressable.at(-4) === 0x00 &&
-				decompressable.at(-3) === 0x00 &&
-				decompressable.at(-2) === 0xff &&
-				decompressable.at(-1) === 0xff;
+				this.strategy.options.compression === CompressionMethod.ZstdNative ||
```

Comment:
I know this is a draft, but flagging now

This check isn't correct, because there's cases where we don't uphold the compression option passes (i.e. the logic in the constructor). I think generally the other code relies on the presence of the Inflate objects specific to that method.

Reply:
Ohhh okay, I didn't notice you shared the `nativeInflate` prop, also forgot what outter if with the getter, I see.

Yeah, I guess this works fine.